src/engraving/libmscore/measure.cpp

```diff
@@ -3968,16 +3968,40 @@ void Measure::checkTrailer()
 }
 
 //---------------------------------------------------------
-//   setStretchedWidth
+//   checkMinWidth
+//    checks that the measure width is larger than the
+//    minimum settings and increases if needed
 //---------------------------------------------------------
 
-void Measure::setStretchedWidth(qreal w)
+double Measure::checkMinWidth(double w)
 {
-    qreal minWidth = isMMRest() ? score()->styleMM(Sid::minMMRestWidth) : score()->styleMM(Sid::minMeasureWidth);
-    if (w < minWidth) {
-        w = minWidth;
+    const double minWidth = isMMRest() ? score()->styleMM(Sid::minMMRestWidth) : score()->styleMM(Sid::minMeasureWidth);
+
+    Segment* s = nullptr;
+    double newSegWidth = 0.0;
+    double diff = 0.0;
+    int iter = 0;
+    double x = 0.0;
+    double multiplier = 1.1; // Empirical factor for fast convergence of the following loop
+
+    while (w < minWidth && iter < 200) { // Just for safety. In practice, it rarely takes more than 2 iter
```

Comment:
This algorithm looks too overcomplicated. It does a lot of things at the same time. Ideally, we should have something like this:

```
void EngravingItem::setWidth(w) {
m_width = std::max(w, minWidth())
}
```
And this is roughly what the previous implementation looked like

Reply:
Sorry for my late reply. Yes, that method was messy, sorry. I think I've now come up with a different, more elegant solution. One that also allows to maintain the "acceptance range" in system justification while accounting for the edge case (code coming soon).

On a more general note though: the previuos implementation wasn't actually simpler, it was hiding the complexity elsewhere. Now, because of many reasons connected to my spacing improvements (I'm happy to go into all details if needed), that specific "elsewhere" is not available anymore, so I need to perform more operations here, making the code a bit more cumbersome. I believe it's a worthy compromise though.